### PR TITLE
Reformat environment variables table

### LINKED ISSUE #4243: Improve tables in documentation
Currently, tables in docs are rendered with a horizontal scroll bar so difficult to read. I think we need to improve this.

e.g. https://docs.cupy.dev/en/stable/reference/environment.html

### PR DESCRIPTION
**Pull request by kmaehashi**
Closes #4243

### CODE REVIEW

**Comment by emcastillo** [comment]
Much cleaner!

File: docs/source/reference/environment.rst [hunk 1]
```diff
@@ -110,26 +111,30 @@ For installation
...
+  Define the compiler to use when compiling CUDA files.
+
+``CUPY_INSTALL_USE_HIP``
+  Default: ``0``
+
+  Build CuPy for AMD ROCm Platform (experimental).
+  For building the ROCm support, see :ref:`install_hip` for further detail.
+
+``CUPY_NVCC_GENERATE_CODE``
+
```
Comment (leofang) [inline docs/source/reference/environment.rst:130]: This empty line should be removed to make it render correctly:
```suggestion
```

**Review by leofang — CHANGES_REQUESTED** [review]

File: docs/source/reference/environment.rst [hunk 2]
```diff
@@ -5,100 +5,101 @@ Environment variables
...
+  The amount of memory that can be allocated for each device.
+  The value can be specified in absolute bytes or fraction (e.g., ``"90%"``) of the total memory of each GPU.
+  See :doc:`../tutorial/memory` for details.
+
+``CUPY_SEED``
+  Set the seed for random number generators.
+
+``CUPY_EXPERIMENTAL_SLICE_COPY``
+  Default: ``0``
```
Comment (kmaehashi) [inline docs/source/reference/environment.rst:54]:
```suggestion
  Default: ``0``
  
```

**Comment by emcastillo** [comment]
Jenkins, test this please

**Review by emcastillo — APPROVED** [review]
Rendered correctly!

**Review by emcastillo — APPROVED** [review]

**Pull request merged**